src/App.java

```diff
@@ -22,17 +24,29 @@ public static void main(String[] args) throws Exception {
     }
 
     public void run() {
-        // initialization
-        initGameControllers();
-        dc.setResizable(false);
-        // main menu
-        MainMenu mainMenu = new MainMenu(dc, gamepads);
-        mainMenu.play();
+        while (running) {
 
-        // clear (currently for testing purposes)
-        dc.clear();
+            // initialization
+            initGameControllers();
+            dc.setResizable(false);
 
-        dc.redraw();
+            // main menu
+            MainMenu mainMenu = new MainMenu(dc, gamepads);
+            mainMenu.play();
+
+            // player select
+            PlayerSelect playerSelect = new PlayerSelect(dc, gamepads);
+            playerSelect.play();
+
+            // define new assigned players array list, this array list should be used in place of the gamepads array list for further screens
+            ArrayList<Gamepad> players = playerSelect.getPlayers();

```

Comment:
Having 2 things named players (as in the controllers of the player and the actual players on the screen) could cause confusion, but we would need to find another name for a player's controller (`playerControllers`?)

Reply:
Yeah, I'm pretty sure we decided on requiring 4 players for all the games so all four players need to select before moving to the next screen (unfortunatly makes it annoying to test). I tested it with four controllers and it works.